Update README.adoc (#8)
in-toto/ITE: ITE/4/README.adoc

authorized functionaries. The security implications of frequently changing
generic resources are discussed in the Security section below.

==== URI Resolvers

> ITE-4 compliant in-toto tools will require a strategy for determining what resources

Reviewer: I'm not very sure if "strategy" is the right word here. I think resolvers are more tangible than the word strategy implies, at least to me. Or perhaps I'm unfamiliar with this particular usage...

Author: I agree strategy is probably not the right word, but I'm unsure what is the right word because its unclear to me what they are. I actually struggled in editing this section, because I don't know if the resolvers are a tool, some added lines of code, some external to the system. If you can give me a better idea of what they are, I can come up with a clearer word to describe them.

Reviewer: Currently, we envision them being separate components that can be plugged in to in-toto tools. I think just "component" in place of "strategy" handles it for now. I'll make a suggestion.

Author: ```suggestion
ITE-4 compliant in-toto tools will require components for determining what resources
```

Author: I agree. Components is better. I made the suggestion, so accept if you like.

Reviewer: I've made suggestions below and this one (and the next line) are at https://github.com/in-toto/ITE/pull/8#discussion_r414883465 and https://github.com/in-toto/ITE/pull/8#discussion_r414883564.

> ITE-4 compliant in-toto tools will require a strategy for determining what resources

Reviewer: ```suggestion
ITE-4 compliant in-toto tools will require pluggable components for determining what resources
```

> these generic URIs are pointing to. Such resolvers, as we refer to these strategies, are required so

Reviewer: ```suggestion
these generic URIs are pointing to. Such resolvers, as we refer to these components, are required so
```

        hashable_representation
```

> `hash_artifacts` closely mirror the general workflow of hashing artifacts in

Reviewer: `hash_artifacts` is the name of one of the functions that resolvers must implement. So while it ends in plural (for artifacts), it's actually a single entity.

Author: Why don't you make that comment a suggestion so I can just accept it here.

Reviewer: Resolved. (I'm unable to mark it for some reason)

Author: OK. That looks good.

Author: Is there something I need to accept?

Reviewer: No, this was taken care of last time with https://github.com/in-toto/ITE/pull/8/commits/6a606fee2ee66d5138a5243dea43ed88b45bdb4c and https://github.com/in-toto/ITE/pull/8/commits/5a401045ee242590896b50c46e697c709412dbd3. I was just going back and trying to mark the ones already fixed as resolved. Same for the other threads where I commented "Resolved".

Author: Ok. By all means remove/ignore any such comments.

> `hash_artifacts` closely mirror the general workflow of hashing artifacts in

Reviewer: ```suggestion
`hash_artifacts` closely mirrors the general workflow of hashing artifacts in
```

> the filesystem, and are similar to the way in-toto reference implementations

Reviewer: ```suggestion
the filesystem, and is similar to the way in-toto reference implementations
```

entities may choose to use a subsection of the contents obtained using the
official API, which returns JSON representations containing different properties
of the entities. It is important for these representations to be consistent for
a particular entity, and to acknowledge the different
sources of non-determinism highlighted in the Appendix. It is also necessary
for this method to handle any missing resources safely. That is, if a resource does not exist

> at the location pointed to by a URI, this method must fail in a secure manner.

Reviewer: Perhaps this should be "fail safely", I'm not sure. @SantiagoTorres thoughts?

Author: "If fail safely" is an accepted phrase, then perhaps we should change it back. Can you make the suggestion?

> at the location pointed to by a URI, this method must fail in a secure manner.

Reviewer: ```suggestion
at the location pointed to by a URI, this method must fail appropriately.
```

I'm mirroring the language in the security section.

Being able to get some hashable representation of abstract entities is also needed

> for in-toto inspections to verify contents. There is no specification mandating

Reviewer: I apologise, I think this is on me, but what I meant was that the in-toto specification doesn't mandate any particular inspections, but leaves it open as a way for supply chain owners to verify anything they need to over and above the regular verification workflow. Therefore, it'd be necessary for them resolve abstract entities into some consistent form that `get_hashable_representation` delivers.

Reviewer 2: I'm not sure if that meaning quite changed, although I may  be missing a crucial part of either sentence.

> how parts of the supply chain are to be verified. Yet, these abstract resources still

Reviewer: ```suggestion
any specific inspections. Yet, these abstract resources still
```

need to be resolvable so that they can be inspected if the supply chain owner so decides to.
Therefore,

> it should be possible for implementations to use get_hashable_representation in scenarios other

Reviewer: get_hashable_respresentation should enclosed in backticks - \`get_hashable_representation\`

Author: Please make the suggestion and I will accept it.

Reviewer: Resolved. (I'm unable to mark it for some reason)

> it should be possible for implementations to use get_hashable_representation in scenarios other

Reviewer: ```suggestion
it should be possible for implementations to use `get_hashable_representation` in scenarios other
```

}
```

The materials are not limited to the commits that make up a pull request. They can
also contain other elements such as reviews, or comments, as well as approvals from maintainers.
It really depends on how the

> implementer, or perhaps GitHub, chooses to define the components of a pull request.

Reviewer: GitHub would be the implementer here.

Author: Again, if you can make your comment a suggestion, I could just accept it.

Reviewer: Resolved. (I'm unable to mark it for some reason)

> implementer, or perhaps GitHub, chooses to define the components of a pull request.

Reviewer: ```suggestion
implementer, perhaps GitHub, chooses to define the components of a pull request.
```
